src/scripts/components/time-slider/time-slider.tsx

```diff
@@ -89,39 +90,83 @@ const TimeSlider: FunctionComponent = () => {
     };
   };
 
+  const outputPosition = Number(
+    ((time - combined.min) * 100) / (combined.max - combined.min)
+  );
+  const inputStyles = cx(
+    styles.input,
+    rangeMain && rangeCompare && styles.compareInput
+  );
+  const compareStyles = cx(
+    styles.compareTrack,
+    rangeMain && rangeCompare && styles.compareBothTracks
+  );
+
   return (
     <div className={styles.timeSlider}>
       <div className={styles.container}>
-        <div className={styles.label}>
-          <div className={styles.labelMin}>{format(combined.min)}</div>
-          <div>{format(time)}</div>
-          <div className={styles.labelMax}>{format(combined.max)}</div>
-        </div>
-
-        <input
-          className={styles.input}
-          type="range"
-          value={time}
-          onChange={({target}) => {
-            const newTime = parseInt(target.value, 10);
-            setTime(newTime);
-            debouncedSetGlobeTime(newTime);
-          }}
-          min={combined.min}
-          max={combined.max}
-          step={stepSize}
-        />
-
         <div className={styles.ranges}>
+          <input
+            className={inputStyles}
+            type="range"
+            value={time}
+            onChange={({target}) => {
+              const newTime = parseInt(target.value, 10);
+              setTime(newTime);
+              debouncedSetGlobeTime(newTime);
+            }}
+            min={combined.min}
+            max={combined.max}
+            step={stepSize}
+          />
+          <output
+            className={styles.timeOutput}
+            style={{
+              left: `${outputPosition}%`
+            }}>
+            {format(time)}
+          </output>
           {rangeMain && (
-            <div
-              className={styles.rangeMain}
-              style={getRangeStyle(rangeMain.min, rangeMain.max)}></div>
+            <div className={styles.mainTrack}>
+              <div
+                className={styles.rangeMain}
+                style={getRangeStyle(rangeMain.min, rangeMain.max)}>
+                {rangeMain.timestamps.map(timestamp => {
```

Comment:
Please calculate the ticks outside so that the jsx markup is more readable. You can pre-calculate the positions in an array and iterate over it here instead of the timestamps.

Reply:
I moved the calculation to a function, like we did it for the range style. Is that fine too?